### PR TITLE
Reimplement the keys method in WSGI CarrierGetter

### LINKED ISSUE #376: WSGI Carrier Getter always returns an empty list as keys
**Describe your environment**
Not relevant.

**Steps to reproduce**
1. Call the method: https://github.com/open-telemetry/opentelemetry-python-contrib/blob/f8e51c488da1f5ab628ba4f5bf25e65638974c44/instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py#L91

**What is the expected behavior?**
I would expect all the keys from the provided carrier returned.

**What is the actual behavior?**
No matter of the carrier contents an empty list is returned.

**Additional context**
Due to this behavior other extensions from this repository might not work as expected.
For example [OT Trace Propagator](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/f8e51c488da1f5ab628ba4f5bf25e65638974c44/propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagators/ot_trace/__init__.py#L92) cannot extract baggage information from the carrier, because empty list is always returned.

### PR DESCRIPTION
**Pull request by marcinzaremba**
# Description

Implements keys method in WSGI CarrierGetter (previously a static, empty list was returned)

Fixes #376 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

See the unit test added in this PR.

# Does This PR Require a Core Repo Change?

- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py [hunk 1]
```diff
@@ -89,7 +91,11 @@ def get(
         return None
 
     def keys(self, carrier):
-        return []
+        return [
+            key[_CARRIER_KEY_PREFIX_LEN:].lower().replace("_", "-")
+            for key in carrier
+            if key.startswith(_CARRIER_KEY_PREFIX)
+        ]
```
Comment (owais) [inline instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py:98]: Out of curiosity, did this prevent or limit propagation in some way when it was implemented as an empty list?

Comment (marcinzaremba) [reply instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py:98]: It prevents any use case when information to be extracted is not known beforehand (for example HTTP headers starting with a particular prefix, but with unknown exact names). The great example is OT Trace Propagator, where the current implementation (empty list) renders injecting baggage unusable: https://github.com/open-telemetry/opentelemetry-python-contrib/blob/f8e51c488da1f5ab628ba4f5bf25e65638974c44/propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagators/ot_trace/__init__.py#L92

Comment (owais) [reply instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py:98]: >  The great example is OT Trace Propagator, where the current implementation (empty list) renders injecting baggage unusable: 

Thanks. This is what I was interested in.

**Review by owais — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
👍

**Pull request merged**